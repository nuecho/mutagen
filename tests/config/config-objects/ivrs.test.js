importDependencies("gvpCustomers-dependencies-config.json");
cfgObjectTests("ivrs", {checkUnchangeableProperties: true});
