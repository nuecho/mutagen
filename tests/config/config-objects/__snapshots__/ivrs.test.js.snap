// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen ivrs config import-export] should create the ivrs: the ivrs have been created 1`] = `
Object {
  "stderr": "",
  "stdout": "+ appprototype [IVRServer-AppPrototype]
+ application [IVRServer-Application]
+ ivr [cfgObjectTest]
Completed. 3 objects created. 0 object updated. 0 object skipped.
",
}
`;

exports[`[mutagen ivrs config import-export] should skip the same ivrs: the ivrs have not been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "= application [IVRServer-Application]
= appprototype [IVRServer-AppPrototype]
= ivr [cfgObjectTest]
Completed. 0 object created. 0 object updated. 3 objects skipped.
",
}
`;

exports[`[mutagen ivrs config import-export] should throw exception when new ivrs to import miss mandatory properties ivrs: exception has been thrown 1`] = `
Object {
  "stderr": "Validation failed.
Missing properties:
  - ivr [invalid]:
    - tenant
    - type
    - version
",
  "stdout": "",
}
`;

exports[`[mutagen ivrs config import-export] should throw exception when updating ivrs with different values for unchangeable properties: exception has been thrown 1`] = `
Object {
  "stderr": "Validation failed.
Unchangeable properties:
  - ivr [cfgObjectTest]:
    - tenant
    - type
",
  "stdout": "",
}
`;

exports[`[mutagen ivrs config import-export] should update the modified ivrs: the modified ivrs have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "~ ivr [cfgObjectTest]
Completed. 0 object created. 1 object updated. 0 object skipped.
",
}
`;
